src/uris/deepLinks/deepLinkService.ts

```diff
@@ -119,35 +143,61 @@ export class DeepLinkService implements Disposable {
 		return undefined;
 	}
 
-	private async processDeepLink(): Promise<void> {
+	private async showOpenRepoPrompt(): Promise<DeepLinkRepoOpenAction> {
+		const result = await window.showInformationMessage(
+			`No matching repository found. Please choose an option to open the repository.`,
+			{ modal: true },
+			{ title: DeepLinkRepoOpenAction.OpenInCurrentWindow },
+			{ title: DeepLinkRepoOpenAction.OpenInNewWindow },
+			{ title: DeepLinkRepoOpenAction.AddToWorkspace },
+			{ title: DeepLinkRepoOpenAction.Cancel, isCloseAffordance: true },
+		);
+
+		return result?.title ?? DeepLinkRepoOpenAction.Cancel;
+	}
+
+	private async processDeepLink(
+		initialAction: DeepLinkServiceAction = DeepLinkServiceAction.DeepLinkEventFired,
+	): Promise<void> {
 		let message = '';
+		let action = initialAction;
+
+		// Remote match
 		let matchingRemotes: GitRemote[] = [];
-		let action: DeepLinkServiceAction = DeepLinkServiceAction.DeepLinkEventFired;
+		let remoteDomain = '';
+		let remotePath = '';
+
+		// Repo open
+		let repoOpenAction = DeepLinkRepoOpenAction.Cancel;
+		let repoOpenUri: Uri | undefined = undefined;
+
 		while (true) {
 			this._context.state = deepLinkStateTransitionTable[this._context.state][action];
 			const { state, repoId, repo, uri, remoteUrl, remote, targetSha, targetType } = this._context;
 			switch (state) {
 				case DeepLinkServiceState.Idle:
 					if (action === DeepLinkServiceAction.DeepLinkErrored) {
 						void window.showErrorMessage('Unable to resolve link');
-						Logger.warn(`Unable to resolve link - ${message}: ${uri?.toString()}`);
+						Logger.warn(`Unable to resolve link - ${message}: ${uri}`);
 					}
 
 					// Deep link processing complete. Reset the context and return.
 					this.resetContext();
 					return;
 				case DeepLinkServiceState.RepoMatch:
 				case DeepLinkServiceState.AddedRepoMatch:
-					if (!repoId) {
+					if (!repoId || !remoteUrl) {
 						action = DeepLinkServiceAction.DeepLinkErrored;
-						message = 'No repo id was provided.';
+						message = 'No repo id or remote url was provided.';
 						break;
 					}
 
+					[, remoteDomain, remotePath] = parseGitRemoteUrl(remoteUrl);
 					// Try to match a repo using the remote URL first, since that saves us some steps.
 					// As a fallback, try to match using the repo id.
 					for (const repo of this.container.git.repositories) {
-						matchingRemotes = await repo.getRemotes({ filter: r => r.url === remoteUrl });
+						// eslint-disable-next-line no-loop-func
+						matchingRemotes = await repo.getRemotes({ filter: r => r.matches(remoteDomain, remotePath) });
```

Comment:
Not a big deal, but could avoid the eslint error by extracting the filter function to be outside the loop: `const remoteFilter = (r: GitRemote) => r.matches(remoteDomain, remotePath);` and then just use it here. That way we avoid creating new closures for each iteration of the loop.

Reply:
I still get `no-loop-func` in that case because I'm in a bigger `while` loop. Also, it gets even worse because I'm declaring a variable in a case statement and get `no-case-declarations`. Despite the new closures, I think this might be the cleanest way to handle it for now.